spec/unit/puppet/type/windowsfeature/windowsfeature_spec.rb

```diff
@@ -3,16 +3,18 @@
 require 'spec_helper'
 
 describe Puppet::Type.type(:windowsfeature) do
-  let :provider_class do
-    described_class.provide(:fake) { mk_resource_methods }
-  end
-  let(:provider) { provider_class.new }
   before do
-    described_class.stubs(:defaultprovider).returns provider_class
-    described_class.stubs(:provider).returns provider
+    @class = described_class
+    @provider_class = @class.provide(:fake) { mk_resource_methods }
```

Comment:
@bastelfreak I reverted this line because I get spec failures when I use this:

```
1) Puppet::Type::Windowsfeature when validating the keywords property installmanagementtools accepts a boolean
     Failure/Error: expect { described_class.new(name: 'feature-name', installmanagementtools: true) }.not_to raise_error

       expected no Exception, got #<Puppet::DevError: Munging failed for value :fake in class provider: undefined method `new' for (provider=fake):Puppet::Type::Windowsfeature::ProviderFake> with backtrace:
         # ./spec/unit/puppet/type/windowsfeature/windowsfeature_spec.rb:36:in `new'
         # ./spec/unit/puppet/type/windowsfeature/windowsfeature_spec.rb:36:in `block (5 levels) in <top (required)>'
         # ./spec/unit/puppet/type/windowsfeature/windowsfeature_spec.rb:36:in `block (4 levels) in <top (required)>'
     # ./spec/unit/puppet/type/windowsfeature/windowsfeature_spec.rb:36:in `block (4 levels) in <top (required)>'
```

Am I missing something? Why the new mocking over the old one? 👍